src/Normalizer/EntityReferenceRevisionItemNormalizer.php

```diff
@@ -50,7 +50,7 @@ protected function constructValue($data, $context) {
       if ($id = $entity->id()) {
         return [
           'target_id' => $id,
-          'target_revision_id' => $entity->getRevisionId(),
+          'target_revision_id' => $data['target_revision_id'],
```

Comment:
The problem here is that if 2 sites have the same revision with different revision ids this won't work as expected. This points to an issue in the entity resolver and may require a new resolver to be written to account for this since we don't have revision_uuids.